app/code/Magento/Catalog/Model/Indexer/Product/Flat/TableBuilder.php

```diff
@@ -46,17 +48,20 @@ class TableBuilder
      * @param \Magento\Catalog\Helper\Product\Flat\Indexer $productIndexerHelper
      * @param \Magento\Framework\App\ResourceConnection $resource
      * @param BuilderInterfaceFactory|null $tableBuilderFactory
+     * @param MetadataPool|null $metadataPool
      */
     public function __construct(
         \Magento\Catalog\Helper\Product\Flat\Indexer $productIndexerHelper,
         \Magento\Framework\App\ResourceConnection $resource,
-        BuilderInterfaceFactory $tableBuilderFactory = null
+        BuilderInterfaceFactory $tableBuilderFactory = null,
+        MetadataPool $metadataPool = null
     ) {
         $this->_productIndexerHelper = $productIndexerHelper;
         $this->resource = $resource;
         $this->_connection = $resource->getConnection();
-        $this->tableBuilderFactory = $tableBuilderFactory ?: \Magento\Framework\App\ObjectManager::getInstance()
+        $this->tableBuilderFactory = $tableBuilderFactory ?: ObjectManager::getInstance()
             ->get(BuilderInterfaceFactory::class);
+        $this->metadataPool = $metadataPool ?: ObjectManager::getInstance()->get(MetadataPool::class);
```

Comment:
Hello @SilinMykola 
Please use null-coalescing operator in constructor body for properties which can be null
```suggestion
        $this->tableBuilderFactory = $tableBuilderFactory ?? ObjectManager::getInstance()
            ->get(BuilderInterfaceFactory::class);
        $this->metadataPool = $metadataPool ?? ObjectManager::getInstance()->get(MetadataPool::class);
```

Reply:
Hello @andrewbess 
Thanks for your review. I made changes, please check.

Reply:
Hello @andrewbess 
Thanks for your review. I made changes, please check.